src/derivative_utils.jl

```diff
@@ -908,42 +873,36 @@ function build_J_W(alg, u, uprev, p, t, dt, f::F, ::Type{uEltypeNoUnits},
         else
             deepcopy(f.jac_prototype)
         end
-        __f = if IIP
-            (du, u, p, t) -> _f(du, u, p, t)
+        W = if J isa StaticMatrix && alg isa OrdinaryDiffEqRosenbrockAdaptiveAlgorithm
```

Comment:
why is this Rosenbrock specific?

Reply:
not sure. I was vaguely following how this worked previously